AppBuilder/platform/views/viewComponent/ABViewReportsManagerComponent.js

```diff
@@ -85,10 +84,10 @@ module.exports = class ABViewReportsManagerComponent extends ABViewComponent {
          async getModels() {
             const reportModels = {};
 
-            if (baseView.settings.datacollectionIDs.length === 0)
-               return reportModels;
-
-            baseView.settings.datacollectionIDs.forEach((dcID) => {
+            (settings.datacollectionIDs.length > 0
+               ? settings.datacollectionIDs
+               : self.view.application.datacollectionIDs
+            ).forEach((dcID) => {
```

Comment:
Can you add a comment explaining? 
The logic looks fine, just need it to be more readable for the future.